### PR TITLE
Glue: Add DB migration & support provisioning for user-defined correlations config

### LINKED ISSUE #55374: Glue: Add DB migration & support provisioning for user-defined correlations config
a migration should be added to `pkg/services/sqlstore/migrations/correlations_mig.go` to add a `config` (name not decided yet) column holding a JSON blob.

Additionally, provisioning datasources/correlations should correctly populate this column

### PR DESCRIPTION
**Pull request by ifrost**
**What this PR does / why we need it**:

This is to add support for adding config object for correlations. At this stage it can only be done with provisioning, frontend still creates no config, but it will change in #55369.

**Which issue(s) this PR fixes**:

Fixes #55374

**Special notes for your reviewer**:

## DB changes

`correlation` table after changes:

```sql
> sqlite3 data/grafana.db ".schema correlation"      

CREATE TABLE `correlation` (
`uid` TEXT NOT NULL
, `source_uid` TEXT NOT NULL
, `target_uid` TEXT NULL
, `label` TEXT NOT NULL
, `description` TEXT NOT NULL
, `config` TEXT NULL, PRIMARY KEY ( `uid`,`source_uid` ));
CREATE INDEX `IDX_correlation_uid` ON `correlation` (`uid`);
CREATE INDEX `IDX_correlation_source_uid` ON `correlation` (`source_uid`);
```

`devenv` provisioned configuration:

```sql
> sqlite3 data/grafana.db "select * from correlation"

6-N91-n4z|PDDA8E780A17E7EF1|gdev-jaeger|Jaeger traces|Related traces stored in Jaeger|
zBHr1an4z|PDDA8E780A17E7EF1|gdev-zipkin|Zipkin traces|Related traces stored in Zipkin|
ifH91-nVk|PDDA8E780A17E7EF1|gdev-prometheus|Logs to metrics|Related metrics stored in Prometheus|{"field":"labels","target":{"expr":"{ job=\"test\" }"}}
```

## Provisioning

```yaml
  - name: gdev-loki
    correlations:
      - targetUID: gdev-jaeger
        label: "Jaeger traces"
        description: "Related traces stored in Jaeger"
      - targetUID: gdev-zipkin
        label: "Zipkin traces"
        description: "Related traces stored in Zipkin"
      - targetUID: gdev-prometheus
        label: "Logs to metrics"
        description: "Related metrics stored in Prometheus"
        config:
          target:
            expr: "{ job=\"test\" }"
          field: "labels"
```

`field` and `target` are mandatory. For backwards compatibility we still allow empty config object. We can stop supporting empty configs once #55369 is implemented.

## Fronted behavior

At the moment when creating a new correlation no config is passed in the payload. API returns empty config object for correlations with no config. This will change in #55369.

Screen Shot 2022-09-21 at 21 59 37

### CODE REVIEW

File: pkg/tests/api/correlations/correlations_create_test.go [hunk 1]
```diff
@@ -211,7 +211,7 @@ func TestIntegrationCreateCorrelation(t *testing.T) {
 		require.NoError(t, res.Body.Close())
 	})
 
-	t.Run("Should correctly create a correlation", func(t *testing.T) {
+	t.Run("Should correctly create a correlation without a config", func(t *testing.T) {
 		description := "a description"
```
Comment (gelicia) [inline pkg/tests/api/correlations/correlations_create_test.go:215]: Did we want to test that config is null here?

File: public/api-merged.json [hunk 2]
```diff
@@ -553,7 +553,7 @@
         "tags": [
           "ldap_debug"
         ],
-        "summary": "Returns the current state of the LDAP background sync integration",
```
Comment (gelicia) [inline public/api-merged.json:556]: What are all the other changes for? They seem fine but not related to the PR

**Review by gelicia — APPROVED** [review]
- Verified a config column is now in the database
- Verified creating a new correlation via the UI creates a config with an empty field and target

Have a couple comments - everything else looks good, but go is not a strong area of mine, a review by someone who knows more go might be prudent.

File: public/api-merged.json (hunk 2 above)
Comment (ifrost) [reply public/api-merged.json:556]: These are currently being generated in `main` as well. This file is generated automatically but is not part of the build process. It's required to run it manually. It looks like it was not run for some changes that are already in `main`.

File: pkg/tests/api/correlations/correlations_create_test.go (hunk 1 above)
Comment (ifrost) [reply pkg/tests/api/correlations/correlations_create_test.go:215]: Yes, it's the current frontend behavior (config is not passed but we want to create a default config object so it's still possible to create correlations in the frontend).

File: public/api-merged.json (hunk 2 above)
Comment (ifrost) [reply public/api-merged.json]: This is cleaned up now after https://github.com/grafana/grafana/pull/55750 got merged and by regenerating swagger with the fix https://github.com/grafana/grafana-enterprise/pull/3895

**Pull request merged**